### PR TITLE
add create index to nodejs client

### PR DESCRIPTION
**Pull request by gsilvestrin**
The api looks like this

```javascript
      const table = await con.openTable('vectors')
      await table.create_index({ type: 'ivf_pq', column: 'vector', num_partitions: 10, max_iters: 2 })
```

In the future we will have other types such as diskann.

### CODE REVIEW

File: rust/vectordb/src/table.rs [hunk 1]
```diff
@@ -83,6 +85,23 @@ impl Table {
...
+    pub async fn create_index_ivf(&self, metric_type: MetricType, num_bits: u8, num_partitions: usize, num_sub_vectors: usize, use_opq: bool, metric_type, max_iterations: usize) {
+        let params: &dyn IndexParams = &VectorIndexParams::ivf_pq(num_partitions, num_bits, num_sub_vectors, use_opq, metric_type, max_iterations);
+        self.dataset.create_index(&*[VECTOR_COLUMN_NAME],IndexType::Vector, None, params)
+    }
+    pub async fn create_index_diskann(&self, /* params here */) {}
+
+    // Option 2 - Single API, accepts IndexParams defined in the lance crate
+    pub async fn create_index(&self, params: &dyn IndexParams) {
+        self.dataset.create_index(&*[VECTOR_COLUMN_NAME],IndexType::Vector, None, params)
+    }
```
Comment (wjones127) [inline rust/vectordb/src/table.rs:102]: @gsilvestrin  What if the API were like:

```js
await table.create_index() // Returns "IndexBuilderSelector" or something
  .ivf({metric_type: my_metric_type, ...})
  .execute();

# or maybe just 
await table.create_index()
  .diskann({...});
```

File: node/src/index.ts [hunk 2]
```diff
@@ -113,8 +113,67 @@ export class Table {
...
+
+  type: 'ivf'
+}
+
+export enum MetricType {
+  L2 = 'l2',
+  Cosine = 'cosine'
+}
+
+export type VectorIndexParams = IvfIndexParamsConfig
```
Comment (eddyxu) [inline node/src/index.ts:175]: Does this mean that it only supports `IvfPQ`? How should it take diskann parameters?

File: rust/vectordb/src/index/mod.rs [hunk 3]
```diff
@@ -0,0 +1 @@
+pub mod vector;
```
Comment (eddyxu) [inline rust/vectordb/src/index/mod.rs:1]: New rust does not need a mod file,

Just put this line into `src/index.rs`

**Review by eddyxu — APPROVED** [review]
Pending CI

**Pull request merged**